### PR TITLE
lightning(dm): coordinate multiple DMWorker

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #3510 

### What is changed and how it works?

use DM-master's etcd to coordinate DM-workers

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
/cc @GMHDBJD 
ptal

**Comment by lance6716** [comment]
ptal

File: dm/master/server.go [hunk 1]
```diff
@@ -1670,6 +1670,14 @@ func (s *Server) generateSubTask(
 		return nil, nil, terror.WithClass(err, terror.ClassDMMaster)
 	}
 
+	var firstMode config.LoadMode
+	for _, stCfg := range stCfgs {
+		if firstMode == "" {
+			firstMode = stCfg.LoaderConfig.ImportMode
+		} else if firstMode != stCfg.LoaderConfig.ImportMode {
+			return nil, nil, terror.ErrConfigInvalidLoadMode.Generatef("found two import-mode %s and %s in task config, DM only support one value", firstMode, stCfg.LoaderConfig.ImportMode)
```
Comment (buchuitoudegou) [inline dm/master/server.go:1678]:
```suggestion
			return nil, nil, terror.ErrConfigInvalidLoadMode.Generatef("found two import-mode %s and %s in task config, DM only supports one value", firstMode, stCfg.LoaderConfig.ImportMode)
```

**Review by buchuitoudegou — APPROVED** [review]
lgtm

File: dm/master/scheduler/scheduler.go [hunk 2]
```diff
@@ -919,7 +921,17 @@ func (s *Scheduler) AddSubTasks(latched bool, expectStage pb.Stage, cfgs ...conf
 		return terror.ErrSchedulerSourcesUnbound.Generate(unbounds)
 	}
 
-	// 4. put the configs and stages into etcd.
+	// 4. put the lightning status, configs and stages into etcd.
+	if cfgs[0].LoaderConfig.ImportMode == config.LoadModePhysical {
```
Comment (GMHDBJD) [inline dm/master/scheduler/scheduler.go:925]: Can we ignore if it's increment mode?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
ptal

File: dm/common/common.go [hunk 3]
```diff
@@ -65,7 +65,9 @@ var (
 	// StageValidatorKeyAdapter is used to store the running stage of the validator.
 	// k/v: Encode(source-id, task-name) -> the running stage of the validator.
 	StageValidatorKeyAdapter KeyAdapter = keyHexEncoderDecoder("/dm-master/stage/validator/")
-
+	// LightningCoordinationKeyAdapter is used to store the coordination information of lightning.
+	// k/v: Encode(task-name, source-id) -> lightning status.
+	LightningCoordinationKeyAdapter KeyAdapter = keyHexEncoderDecoder("/dm-master/lightning/")
```
Comment (lichunzhu) [inline dm/common/common.go:70]: Is `"/dm-master/lightning/"` too general? How about `"/dm-master/lightning-coord/"` or ``"/dm-master/lightning-status/"`

**Review by lichunzhu — COMMENTED** [review]
rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**